jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -81,6 +81,38 @@ static void verifyImageMetadata(ImageMetadataTemplate metadata, Path metadataCac
     this.cacheStorageFiles = cacheStorageFiles;
   }
 
+  /**
+   * Returns {@code true} if all image layers described in a manifest have a corresponding file
+   * entry in the cache.
+   *
+   * @param manifest the image manifest
+   * @return a boolean
+   */
+  boolean allLayersCached(ManifestTemplate manifest) {
```

Comment:
Would it possible to move this method directly to PullBaseImageStep? Since that is the only class where this is used and it could help us avoid having to go through an extra layer of classes to reach this method. Or does this method require any variables that are specific to `CacheStorageReader`?

Reply:
Yeah, I was thinking about this too - this method goes through a few layers because it depends on `CacheStorageReader`’s `cacheStorageFiles` variable [here](https://github.com/GoogleContainerTools/jib/pull/3767/files/ac8a2c7f45bb227e9f35f373510356c524c0ba7c#diff-faacb2e3a1c5067517059fc35d48a2de6de8483ee3d126f79c5f89cbc7e6e780R108).

Reply:
Ah you’re right - I missed that we could just add `cacheStorageFiles` as a field and move this logic to `Cache`. 

I went ahead and tried to make this change, but noticed that the `Cache` class is set up in a way that many of its methods are wrappers around calls to either `CacheStorageReader` or `CacheStorageWriter`. Looking at the existing test suites, `CacheStorageReaderTest` is also more straightforward to add to than `CacheTest` for unit testing the new check. 

I am tempted to leave this logic in `CacheStorageReader` just to stay consistent with the existing setup here - lmk what you think!